From a07008fbecdf65ef7ffd9b90ae3a5c0efb422d46 Mon Sep 17 00:00:00 2001
Message-Id: <a07008fbecdf65ef7ffd9b90ae3a5c0efb422d46.1497521630.git.jstancek@redhat.com>
From: Cyril Hrubis <chrubis@suse.cz>
Date: Wed, 14 Jun 2017 17:25:09 +0200
Subject: [PATCH] configure.ac: Generate linux_syscall_headers.h

This avoids a race in generating the linux_syscall_headers.h on a
parallel build with many CPUs.

Note that the race has been reduced in commit:

commit 3f385652efe811fe7491474f8513baf44cf0a12d
Author: Cyril Hrubis <chrubis@suse.cz>
Date:   Wed Jan 13 14:13:31 2016 +0100

    build: regen.sh (linux_syscall_numbers.h): Fix race

But was not completely fixed.

After discussing the problem the easiest solution is to generate the
header before the build processes starts otherwise we would have to make
complex changes to the LTP build system. So this commits piggy backs the
regen.sh script invocation at the end of the configure script.

Fixes #161

Signed-off-by: Cyril Hrubis <chrubis@suse.cz>
Acked-by: Jan Stancek <jstancek@redhat.com>
---
 configure.ac | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure.ac b/configure.ac
index ecc9a2699109..cbe01d34f681 100644
--- a/configure.ac
+++ b/configure.ac
@@ -132,6 +132,9 @@ fi
 
 AC_CONFIG_SUBDIRS([utils/ffsb-6.0-rc2])
 
+AC_CONFIG_COMMANDS([linux_syscall_headers.h],
+		   [cd testcases/kernel/include; ./regen.sh])
+
 # END testsuites knobs
 LTP_CHECK_FORTIFY_SOURCE
 LTP_CHECK_CC_WARN_OLDSTYLE
-- 
1.8.3.1

